File: backend/src/main/java/com/elhabhab/backend/dto/response/OrderResponseDTO.java
package com.elhabhab.backend.dto.response;


import com.elhabhab.backend.enums.OrderStatus;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OrderResponseDTO {
    private UUID orderId;
    private UUID userId;
    private Long addressId;
    private List<OrderItemResponseDTO> items;
    private BigDecimal totalAmount;
    private LocalDateTime orderDate;
    private OrderStatus status;

    private String couponCode;
    private BigDecimal discountAmount;
}

File: frontend-angular/src/app/interface/order-response-dto.ts
import { OrderItemResponse } from "./order-item-response";

export interface OrderResponseDTO {
  orderId: string;
  userId: string;
  addressId : number;
  items: OrderItemResponse[];
  totalAmount: number;
  orderDate: string;
  status: string;
  couponCode?: string;
  discountAmount?: number;
}
